community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -380,15 +383,40 @@ public RelationshipIterator nodeGetRelationships( long nodeId, Direction directi
         }
     }
 
+    private static int[] deduplicate( int[] types )
+    {
+        int unique = 0;
+        for ( int i = 0; i < types.length; i++ )
+        {
+            int type = types[i];
+            for ( int j = 0; j < unique; j++ )
+            {
+                if ( type == types[j] )
+                {
+                    type = -1; // signal that this relationship is not unique
+                    break; // we will not find more than one conflict
+                }
+            }
+            if ( type != -1 )
+            { // this has to be done outside the inner loop, otherwise we'd never accept a single one...
+                types[unique++] = types[i];
+            }
+        }
+        if ( unique < types.length )
+        {
+            types = Arrays.copyOf( types, unique );
+        }
+        return types;
+    }
```

Comment:
We could alternatively sort the types, and do the uniqueness check in a single pass. _Iff_ it could help us to know that the types are sorted up front, then this would be a good idea. I haven't checked if this premise is true, but it sounds like a logical thing.